vault/resource_aws_auth_backend_roletag_blacklist.go

```diff
@@ -0,0 +1,153 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	awsAuthBackendRoleTagBlacklistBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/config/tidy/roletag-blacklist$")
+)
+
+func awsAuthBackendRoleTagBlacklistResource() *schema.Resource {
+	return &schema.Resource{
+		Create: awsAuthBackendRoleTagBlacklistWrite,
+		Read:   awsAuthBackendRoleTagBlacklistRead,
+		Update: awsAuthBackendRoleTagBlacklistWrite,
+		Delete: awsAuthBackendRoleTagBlacklistDelete,
+		Exists: awsAuthBackendRoleTagBlacklistExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
```

Comment:
Can you expand on the usage of this resource? It strikes me as odd that it has an optional `backend`. The usage in the test config imply "use this roletag blacklist on this backend", which I see as "I have a backend, I would like to also have this roletag blacklist attached to _this specific backend_". The default / optional value here makes me think that this resource could be used without a backend declared at all. Could the two resources be created like so?:

```
resource "vault_auth_backend" "aws" {
  path = "aws123"
  type = "aws"
}

resource "vault_aws_auth_backend_roletag_blacklist" "test" {
  backend = "aws123"
  safety_buffer = 8600
  disable_periodic_tidy = true
}
```

Is this config alone valid?

```
resource "vault_aws_auth_backend_roletag_blacklist" "test" {
  backend = "aws123"
  safety_buffer = 8600
  disable_periodic_tidy = true
}
```

The optional + default for `backend` implies to me this resource can stand alone and function without a `vault_auth_backend` to operate on, but reading the Vault docs I felt otherwise. What am I missing?

Reply:
That first config is the correct usage. The second config is invalid _unless_ you've already configured the auth backend outside of Terraform. I originally went with the optional+default because that's how the Vault API functions when mounting a backend; it just assumes a default value. And in my head, for whatever reason, carrying that behaviour over here made sense, even though now that we're talking about it, it seems much more intuitive that it be required to be explicit. TL;DR: I agree with you, should be Required instead of Optional+Default.

I'm going to update it here, at the very least. My preference is to also update it everywhere else `backend` is used as a field, to be consistent. Do you have a strong preference either way?